intrastat_product/models/intrastat_product_declaration.py

```diff
@@ -308,16 +308,26 @@ def _get_partner_country(self, inv_line, notedict, eu_countries):
         return country
 
     def _get_intrastat_transaction(self, inv_line, notedict):
+        tr_11 = self.env.ref("intrastat_product.intrastat_transaction_11")
+        tr_12 = self.env.ref("intrastat_product.intrastat_transaction_12")
+        tr_21 = self.env.ref("intrastat_product.intrastat_transaction_21")
+
         invoice = inv_line.move_id
         transaction = invoice.intrastat_transaction_id
-        fp2transaction = {
-            "b2b": self.env.ref("intrastat_product.intrastat_transaction_11"),
-            "b2c": self.env.ref("intrastat_product.intrastat_transaction_12"),
-        }
-        if not transaction:
-            # as we have searched with intrastat_fiscal_position in ('b2b', 'b2c')
-            # we should always have intrastat_fiscal_position in fp2transaction
-            transaction = fp2transaction[invoice.intrastat_fiscal_position]
+        intr_fp = invoice.intrastat_fiscal_position
```

Comment:
Why using fiscal position?

Reply:
Because in fiscal position you can define if it is code 11 or 12:

![imagen](https://github.com/OCA/intrastat-extrastat/assets/46522913/a4880fca-b4fe-49ec-910b-b64e416fc927)

Reply:
OK